src/app/tasks/task-comments-viewer/task-comments-viewer.coffee

```diff
@@ -29,34 +30,55 @@ angular.module("doubtfire.tasks.task-comments-viewer", [])
         project_id: $scope.project.project_id,
         task_definition_id: $scope.task.task_definition_id
       }, (response) ->
-        $scope.task.comments = _.map(response, taskService.mapComment)
+        $scope.task.comments = _.map(response, taskService.mapComment) #in the HTML, the mapped task.comments are displayed
         $scope.task.num_new_comments = 0
         scrollDown()
         $scope.focus?() if $scope.refocusOnTaskChange
 
+        CommentResourceService.setTask($scope.task)
+
     # Automatically scroll the inner div to the bottom of comments
     scrollDown = ->
       $timeout ->
         objDiv = document.querySelector("task-comments-viewer .panel-body")
         wrappedResult = angular.element(objDiv)
         wrappedResult[0].scrollTop = wrappedResult[0].scrollHeight
 
+    $scope.openCommentsModal = (comment)->
+      imageUrl = $sce.trustAsResourceUrl(Task.generateCommentsAttachmentUrl($scope.project, $scope.task, comment))
+      CommentResourceService.setImageUrl(imageUrl)
+      CommentsModal.show()
+
     # Checks for enter keydown
     $scope.enterDown = (editor) ->
       $scope.addComment()
       return CodeMirror.Pass
 
+    #===========================================================================================
+    $scope.canUserEdit = (comment) ->
+      canEdit = false
+      if comment.author_is_me || currentUser.role == "Admin"
+        canEdit = true
+      canEdit
+
+    #===========================================================================================
+    $scope.getCommentAttachment = (comment) ->
+      mediaURL = $sce.trustAsResourceUrl(Task.generateCommentsAttachmentUrl($scope.project, $scope.task, comment))
+      mediaURL
+
+    #===========================================================================================
     # Submits a new comment
     $scope.addComment = ->
       $scope.comment.text = $scope.comment.text.trim()
-      taskService.addComment $scope.task, $scope.comment.text,
-        (success) ->
-          $scope.comment.text = ""
-          analyticsService.event "View Task Comments", "Added new comment"
-          scrollDown()
-        (response) ->
-          alertService.add("danger", response.data.error, 2000)
+      taskService.addComment $scope.task, $scope.comment.text, CommentResourceService.commentType,
```

Comment:
Move selection of text/image/audio/video here... then we know what we need to add. Remove from CommentResourceService

Markdown editor needs to just handle markdown